src/sqlfluff/core/linter/linter.py

```diff
@@ -0,0 +1,760 @@
+            for violation in vs:
+                violation.ignore_if_in(config.get("ignore"))
+
+        linted_file = LintedFile(
+            fname,
+            vs,
+            time_dict,
+            tree,
+            ignore_mask=ignore_buff,
+            templated_file=parsed.templated_file,
+        )
+
+        # This is the main command line output from linting.
+        if self.formatter:
+            self.formatter.dispatch_file_violations(
+                fname, linted_file, only_fixable=fix
+            )
+
+        # Safety flag for unset dialects
+        if config.get("dialect") == "ansi" and linted_file.get_violations(
+            fixable=True if fix else None, types=SQLParseError
+        ):
+            if self.formatter:
+                self.formatter.dispatch_dialect_warning()
+
+        return linted_file
+
+    def paths_from_path(
+        self,
+        path: str,
+        ignore_file_name: str = ".sqlfluffignore",
+        ignore_non_existent_files: bool = False,
+        ignore_files: bool = True,
+        working_path: str = os.getcwd(),
+    ) -> List[str]:
+        """Return a set of sql file paths from a potentially more ambiguous path string.
+
+        Here we also deal with the .sqlfluffignore file if present.
+
+        When a path to a file to be linted is explicitly passed
+        we look for ignore files in all directories that are parents of the file,
+        up to the current directory.
+
+        If the current directory is not a parent of the file we only
+        look for an ignore file in the direct parent of the file.
+
+        """
+        if not os.path.exists(path):
+            if ignore_non_existent_files:
+                return []
+            else:
+                raise IOError("Specified path does not exist")
+
+        # Files referred to exactly are also ignored if
+        # matched, but we warn the users when that happens
+        is_exact_file = not os.path.isdir(path)
+
+        if is_exact_file:
+            # When the exact file to lint is passed, we
+            # fill path_walk with an input that follows
+            # the structure of `os.walk`:
+            #   (root, directories, files)
+            dirpath = os.path.dirname(path)
+            files = [os.path.basename(path)]
+            ignore_file_paths = ConfigLoader.find_ignore_config_files(
+                path=path, working_path=working_path, ignore_file_name=ignore_file_name
+            )
+            # Add paths that could contain "ignore files"
+            # to the path_walk list
+            path_walk_ignore_file = [
+                (
+                    os.path.dirname(ignore_file_path),
+                    None,
+                    # Only one possible file, since we only
+                    # have one "ignore file name"
+                    [os.path.basename(ignore_file_path)],
+                )
+                for ignore_file_path in ignore_file_paths
+            ]
+            path_walk: Union[

```

Comment:
This type hint is fairly complex. Could we use aliases here to make it more readable?

https://docs.python.org/3/library/typing.html#type-aliases

>Type aliases are useful for simplifying complex type signatures. For example:...

Reply:
great idea.